CHANGELOG.md

```diff
@@ -1,8 +1,75 @@
+* Function `toISOWeek` returned wrong results for year 1970. [#4988](https://github.com/yandex/ClickHouse/pull/4988) ([alexey-milovidov](https://github.com/alexey-milovidov))
+* Now follower replicas would not send queries to leader if they receive them from DDL. [#4991](https://github.com/yandex/ClickHouse/pull/4991) ([alesapin](https://github.com/alesapin))
+* Fix wrong result for 'select distinct' with multiple streams (ex. leaded by join) [#5001](https://github.com/yandex/ClickHouse/pull/5001) ([Artem Zuikov](https://github.com/4ertus2))
+* The `max_partitions_per_insert_block` field is of `size_t` type, rather than `Boolean` [#5028](https://github.com/yandex/ClickHouse/pull/5028) ([Mohammad Hossein Sekhavat](https://github.com/mhsekhavat))
+
+### Build/Testing/Packaging Improvement
+* Further improvement of the docker-based test machinery ... [#4713](https://github.com/yandex/ClickHouse/pull/4713) ([Vasily Nemkov](https://github.com/Enmk))
+*  [#4937](https://github.com/yandex/ClickHouse/pull/4937) ([alesapin](https://github.com/alesapin))
+* Tests: Allow use any test database (remove test.hardcode where it possible) [#5008](https://github.com/yandex/ClickHouse/pull/5008) ([proller](https://github.com/proller))
+* Fix ubsan errors [#5037](https://github.com/yandex/ClickHouse/pull/5037) ([Vitaly Baranov](https://github.com/vitlibar))
+
+### Improvement
+* Disable push-down to right table in left join, left table in right join, and both tables in full join. [#4846](https://github.com/yandex/ClickHouse/pull/4846) ([Ivan](https://github.com/abyss7))
+* Copier: auto upload task configuration from --task-file option [#4876](https://github.com/yandex/ClickHouse/pull/4876) ([proller](https://github.com/proller))
+* Added typos handler for storage factory and table functions factory. [#4891](https://github.com/yandex/ClickHouse/pull/4891) ([Danila Kutenin](https://github.com/danlark1))
+* Support asterisks and qualified asterisks for multiple joins without subqueries [#4898](https://github.com/yandex/ClickHouse/pull/4898) ([Artem Zuikov](https://github.com/4ertus2))
+* Make missing column error message more user friendly. [#4915](https://github.com/yandex/ClickHouse/pull/4915) ([Artem Zuikov](https://github.com/4ertus2))
+* Use another allocator for MarkCache and UncompressedCache to catch segfaults faster. I had to move functions to inline definition because otherwise I would need to copy-paste a lot of code, for now the code is fully reused but compilation time can increase a bit [#4928](https://github.com/yandex/ClickHouse/pull/4928) ([Danila Kutenin](https://github.com/danlark1))
+* Deduplicate pointers in ASTSelectQuery and its children cause they could became not consistent. Leave one copy + positions map. [#4952](https://github.com/yandex/ClickHouse/pull/4952) ([Artem Zuikov](https://github.com/4ertus2))
```

Comment:
What bug was fixed?

Reply:
@4ertus2

Reply:
The only known bug with #4952 was error in constants optimisation described here https://github.com/yandex/ClickHouse/issues/4242#issuecomment-477250658
But there could be some unknown ones cause AST was inconsistent after optimisation.